internal/testing/oteltest/diff.go

```diff
@@ -86,7 +86,7 @@ func mapStatusCode(code gcerrors.ErrorCode) codes.Code {
 	return codes.Error
 }
 
-func diffSpans(got []sdktrace.ReadOnlySpan, prefix string, want []Call) string {
+func DiffSpans(got []sdktrace.ReadOnlySpan, prefix string, want []Call) string {
```

Comment:
Instead of exposing these separately (and having 3 overlapping functions in the package), can you keep diffSpans and diffMetrics private, and use Diff from runtimevar tests?

Reply:
@vangent So the challange is Diff is usefull only when you have spans that endup generating metrics, however in this case the metric is generated independent of any spans. We can at minimum keep diffSpans private but DiffMetrics will need to remain public.

Reply:
Alternatively in the diffSpans we can add a check on supplied calls such that if the Call.Method is empty we assume it was not intended to be used for checking Diff on the span. Personally I feel that makes the function unnecessarily convoluted just to avoid exposing required functionality in a test package.

Reply:
OK, let's make `DiffSpans` -> `diffSpans` then, as it sounds like a caller would always use `Diff` instead.

Agreed on not making Diff more complex.